### PR TITLE
v0.25: api keys: guides, settings, options

### LINKED ISSUE #1287: v0.25: API keys: Guides, settings, options
Besides requiring rewriting significant sections of the Security/Authentication guide and the API reference, the new API keys has an effect on many other articles that mention things such as the `master` key.

These include any guides on deploying MeiliSearch in production environments or configuring cloud services, as well as configuration options references.

--- 

## Tasks

- [ ] Search for mentions of previous Authentication system since it is possible the current list is not exhaustive
- [x] reference/features/configuration.md#environment: update references to keys (at a minimum remove references to the `master` key
- [x] `reference/features/configuration.md#master-key`: rewrite
- [x] Update create/how_to/running_production
- [x] create/how_to/postman_collection (not sure about this one)
- [x] create/how_to/aws
- [x] create/how_to/gcp
- [x] create/how_to/digitalocean_droplet
- [x] create/how_to/qovery (this guide is out of date and only compatible with qovery v1.0)
- [x] create/how_to/search_bar_for_docs
- [x] Add a mention to fine-grained API key management to learn/what_is_meilisearch/features

## Reference
- spec PR: https://github.com/meilisearch/specifications/pull/85

### PR DESCRIPTION
**Pull request by maryamsulemani97**
closes #1287

### CODE REVIEW

**Comment by maryamsulemani97** [comment]
I only updated the text from "authentication" to "security" as Gui updated the links in his [PR](https://github.com/meilisearch/documentation/pull/1307)

File: reference/features/README.md [hunk 1]
```diff
@@ -6,7 +6,7 @@ Understanding our reference documentation is easier with background knowledge of
 
 ## Table of contents
 
-[Authentication](/reference/features/authentication.md)
+[Security](/reference/features/authentication.md)
```
Comment (guimachiavelli) [inline reference/features/README.md:9]:
```suggestion
[Security](/reference/features/security.md)
```

File: reference/features/configuration.md [hunk 2]
```diff
@@ -131,7 +131,7 @@ Sets the HTTP address and port MeiliSearch will use.
 **Default value**: `None`
 **Expected value**: an alphanumeric string
 
-Sets the instance's master key, automatically protecting all routes except [`GET /health`](/reference/api/health.md).
+Sets the instance's master key, automatically protecting all routes except [`GET /health`](/reference/api/health.md). It gives access to the [`/keys` endpoint](/reference/api/keys.md) allowing you to create, update, and delete API keys.
```
Comment (guimachiavelli) [inline reference/features/configuration.md:134]:
```suggestion
Sets the instance's master key, automatically protecting all routes except [`GET /health`](/reference/api/health.md). This means you will need an API key to access endpoints such as `POST /search` and `GET /documents`. [You can read more about security keys in MeiliSearch in our dedicated guide.](/reference/features/security.md)
```

File: learn/what_is_meilisearch/features.md [hunk 3]
```diff
@@ -54,6 +54,10 @@ If you enclose search terms in double quotes (`"`), MeiliSearch will only return
 
 Phrase search is particularly useful when looking for a single result, such as searching for a book by ISBN. It is also possible to combine phrase searches with MeiliSearch's basic syntax so only parts of a query are matched strictly.
 
+## API key management
+
+MeiliSearch allows you to protect your instances through the use of API keys. API keys allow you to control which users can access what indexes, routes, and endpoints. These API keys are stored and managed by the master key. When you set a master key on your first launch, MeiliSearch generates two API keys called the `Default Search API Key` and the `Default Admin API Key`. You can read more them in our article on [security](/reference/features/security.md).
```
Comment (guimachiavelli) [inline learn/what_is_meilisearch/features.md:59]: The master key doesn't really store keys, only manages them, right?

```suggestion
MeiliSearch allows you to protect your instances through the use of API keys. API keys allow you to control which users can access what indexes, routes, and endpoints. These API keys are managed by the master key. When you set a master key on your first launch, MeiliSearch generates two API keys called the `Default Search API Key` and the `Default Admin API Key`. You can read more them in our article on [security](/reference/features/security.md).
```

**Review by guimachiavelli — CHANGES_REQUESTED** [review]

File: reference/features/README.md (hunk 1 above)
Comment (maryamsulemani97) [reply reference/features/README.md:9]: You updated all existing the links in your PR, so I didn’t update them here.

File: create/how_to/search_bar_for_docs.md [hunk 4]
```diff
@@ -113,8 +113,8 @@ If you don't want to use Docker, here are [other ways to run the scraper](https:
 
 `<absolute-path-to-your-config-file>` should be the **absolute** path of your configuration file defined at [the previous step](#configuration-file).
 
-The API key you must provide should have the permissions to add documents into your MeiliSearch instance. In a production environment, we recommend providing the private key instead of the master key, as it is safer and it has enough permissions to perform such requests.
-_More about [MeiliSearch authentication](/reference/features/authentication.md)._
+The API key used here should have the permissions to add documents into your MeiliSearch instance. In a production environment, we recommend providing the `Default Admin API Key` instead of the master key, as it is safer and it has enough permissions to perform such requests.
```
[the remark this answers is not in the document]
Comment (maryamsulemani97) [reply create/how_to/search_bar_for_docs.md:116]: 2. Agree with the complexity part

File: create/how_to/running_production.md [hunk 5]
```diff
@@ -103,7 +103,7 @@ In Debian and other Linux distributions, `systemd` allows you to create and mana
...
 
-To run MeiliSearch in a production environment, use the `--env` flag. To generate a master key that will let MeiliSearch create reading and writing keys, use the `--master-key` flag. With those keys, you can easily control who can access or create new documents, indexes, or change the configuration. You can change the `Master Key` to any value in the following command. However, for security concerns, it's better to choose a safe and random key, never share it and, just, **keep it safe**.
+To run MeiliSearch in a production environment, use the `--env` flag. Generate the master key using the `--master-key` flag. When you launch a secured instance for the first time, MeiliSearch creates two default API keys: [`Default Search API Key` and `Default Admin API Key`](/reference/features/security.md#default-api-keys). With the `Default Admin API Key`, you can easily control who can access or create new documents, indexes, or change the configuration. You can change the `Master Key` to any value in the following command. However, for security concerns, it's better to choose a safe and random key, never share it and, just, **keep it safe**.
```
[the remark this answers is not in the document]
Comment (maryamsulemani97) [reply create/how_to/running_production.md:106]: Updated 3 & 5. We can address the others in a separate PR

File: learn/what_is_meilisearch/features.md [hunk 6]
```diff
@@ -54,6 +54,10 @@ If you enclose search terms in double quotes (`"`), MeiliSearch will only return
 
 Phrase search is particularly useful when looking for a single result, such as searching for a book by ISBN. It is also possible to combine phrase searches with MeiliSearch's basic syntax so only parts of a query are matched strictly.
 
+## API key management
+
+MeiliSearch allows you to protect your instances through the use of API keys. API keys allow you to control which users can access what indexes, routes, and endpoints. These API keys are managed by the master key. When you set a master key on your first launch, MeiliSearch generates two API keys called the `Default Search API Key` and the `Default Admin API Key`. You can read more them in our article on [security](/reference/features/security.md).
```
Comment (guimachiavelli) [inline learn/what_is_meilisearch/features.md:59]: A couple minor edits:

1. Avoiding starting two sentences with "[thing] allows you to [do something]";
2. Breaking the paragraph in two for a bit easier reading? (Might be unnecessary, feel free to ignore it if you disagree). 

```suggestion
MeiliSearch allows you to protect your instances through the use of API keys. API keys give you control over which users can access what indexes, routes, and endpoints.

API keys are managed by the master key. When you set a master key on your first launch, MeiliSearch generates two API keys called the `Default Search API Key` and the `Default Admin API Key`. You can read more them in our article on [security](/reference/features/security.md).
```

File: create/how_to/search_bar_for_docs.md [hunk 7]
```diff
@@ -113,8 +113,8 @@ If you don't want to use Docker, here are [other ways to run the scraper](https:
 
 `<absolute-path-to-your-config-file>` should be the **absolute** path of your configuration file defined at [the previous step](#configuration-file).
 
-The API key you must provide should have the permissions to add documents into your MeiliSearch instance. In a production environment, we recommend providing the private key instead of the master key, as it is safer and it has enough permissions to perform such requests.
-_More about [MeiliSearch authentication](/reference/features/authentication.md)._
+The API key should have the permissions to add documents into your MeiliSearch instance. In a production environment, we recommend providing the `Default Admin API Key` instead of the master key, as it is safer and it has enough permissions to perform such requests.
```
Comment (guimachiavelli) [inline create/how_to/search_bar_for_docs.md:116]: What do you think about not mentioning the master key at all here?

```suggestion
The API key should have the permissions to add documents into your MeiliSearch instance. In a production environment, we recommend providing the `Default Admin API Key` as it has enough permissions to perform such requests.
```

File: create/how_to/running_production.md [hunk 8]
```diff
@@ -103,7 +103,7 @@ In Debian and other Linux distributions, `systemd` allows you to create and mana
...
 
-To run MeiliSearch in a production environment, use the `--env` flag. To generate a master key that will let MeiliSearch create reading and writing keys, use the `--master-key` flag. With those keys, you can easily control who can access or create new documents, indexes, or change the configuration. You can change the `Master Key` to any value in the following command. However, for security concerns, it's better to choose a safe and random key, never share it and, just, **keep it safe**.
+To run MeiliSearch in a production environment, use the `--env` flag. Generate the master key using the `--master-key` flag. When you launch a secured instance for the first time, MeiliSearch creates two default API keys: [`Default Search API Key` and `Default Admin API Key`](/reference/features/security.md#default-api-keys). With the `Default Admin API Key`, you can control who can access or create new documents, indexes, or change the configuration. You can change the master Key to any value in the following command. However, for security concerns, it's better to choose a safe and random key, never share it and, just, **keep it safe**.
```
Comment (guimachiavelli) [inline create/how_to/running_production.md:106]:
```suggestion
To run MeiliSearch in a production environment, use the `--env` flag. Generate the master key using the `--master-key` option. When you launch an instance for the first time, MeiliSearch creates two default API keys: [`Default Search API Key` and `Default Admin API Key`](/reference/features/security.md#default-api-keys). With the `Default Admin API Key`, you can control who can access or create new documents, indexes, or change the configuration. 

Remember to choose a safe and random key and avoid exposing it in publicly accessible applications. You can change the master key with the following command:
```

**Review by guimachiavelli — CHANGES_REQUESTED** [review]

**Review by guimachiavelli — APPROVED** [review]
LGTM 🐶

**Pull request merged**